XenModel/XenAPI-Extensions/Network.cs

```diff
@@ -295,6 +295,18 @@ public bool IsVLAN()
 
         public bool IsSriov()
         {
+            if(this.IsVLAN())
+            {
+                foreach (PIF pif in Connection.ResolveAll<PIF>(PIFs))

```

Comment:
PIFs is resolved in IsVLAN() and then again inside this block. It would be nice to avoid this. 
Also the code to check if the vlan is on sr-iov is repeated in NetworkList.cs, so it is suitable to be extracted into a function. And then this function would be simplified to something like: 
`return Connection.ResolveAll(PIFs).Find(pif => pif.IsSriovLogicalPIF() || pif.IsVlanOnSriov) != null;`

Reply:
Thanks. I have changed the logic in Network.cs. But I have no idea to avoid the code repeated in NetworkList.cs because I need to get the logic pif to judge if the sriov network needs host to reboot.

Reply:
How about this:
- instead of PIF.IsVlanOnSriov() we have a method that gives us the Network_sriov,  e.g.
```
        public XenRef<Network_sriov> NetworkSriov()
        {
            if (IsSriovLogicalPIF())
                return sriov_logical_PIF_of[0];

            if (VLAN < 0)
                return null;
            var vlan = Connection.Resolve(VLAN_master_of);
            if (vlan == null)
                return null;
            var taggedPif = Connection.Resolve(vlan.tagged_PIF);
            return taggedPif != null && taggedPif.IsSriovLogicalPIF() ? taggedPif.sriov_logical_PIF_of[0] : null;
        }
```
- then Netwotk.IsSrIov() becomes:
```
return Connection.ResolveAll(PIFs).Any(pif => pif.NetworkSriov() != null);
```
- and in NetworkList we can have:
```
                var networkSriov = Pif != null ? Pif.NetworkSriov() : null;
                SriovCell.Value = networkSriov == null
                    ? Messages.NO
                    : Pif.Connection.Resolve(networkSriov).requires_reboot
                        ? Messages.HOST_NEEDS_REBOOT_ENABLE_SRIOV 
                        : Messages.YES;
```